src/metaxy/metadata_store/postgres.py

```diff
@@ -0,0 +1,257 @@
+
+import logging
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from metaxy.data_versioning.calculators.ibis import HashSQLGenerator
+    from metaxy.metadata_store.base import MetadataStore
+
+from metaxy.data_versioning.hash_algorithms import HashAlgorithm
+from metaxy.metadata_store.ibis import IbisMetadataStore
+
+logger = logging.getLogger(__name__)
+
+
+class PostgresMetadataStore(IbisMetadataStore):
+    """
+    [PostgreSQL](https://www.postgresql.org/) metadata store using [Ibis](https://ibis-project.org/) backend.
+
+    Provides production-grade metadata storage with full ACID compliance, schema isolation,
+    and support for advanced hash algorithms via PostgreSQL extensions.
+
+    **Hash Algorithm Support:**
+
+    - **MD5** (default): Built-in, no extensions required
+    - **SHA256**: Requires `pgcrypto` extension (auto-enabled on open)
+
+    The store automatically enables `pgcrypto` when using SHA256 hashing.
+
+    **Schema Isolation:**
+
+    Use the `schema` parameter to isolate metadata in a dedicated schema.
+    If not specified, tables are created in the user's search_path (typically `public`).
+
+    Example: Connection String
+        ```py
+        store = PostgresMetadataStore("postgresql://user:pass@localhost:5432/metadata")
+        ```
+
+    Example: Connection Parameters
+        ```py
+        store = PostgresMetadataStore(
+            host="localhost",
+            port=5432,
+            user="ml",
+            password="secret",
+            database="metaxy",
+            schema="features",  # Optional schema isolation
+        )
+        ```
+
+    Example: SHA256 Hash Algorithm
+        ```py
+        # pgcrypto extension auto-enabled
+        store = PostgresMetadataStore(
+            "postgresql://user:pass@localhost:5432/metadata",
+            hash_algorithm=HashAlgorithm.SHA256,
+        )
+        ```
+
+    Example: With Fallback Stores
+        ```py
+        # Read from prod, write to dev
+        prod_store = PostgresMetadataStore(
+            "postgresql://user:pass@prod:5432/metadata",
+        )
+        dev_store = PostgresMetadataStore(
+            "postgresql://user:pass@dev:5432/metadata",
+            fallback_stores=[prod_store],
+        )
+        ```
+
+    Warning:
+        SHA256 requires the `pgcrypto` extension. The store will attempt to enable it
+        automatically on open. Ensure your database user has CREATE EXTENSION privileges,
+        or have a DBA pre-create the extension.
+    """
+
+    def __init__(
+        self,
+        connection_string: str | None = None,
+        *,
+        host: str | None = None,
+        port: int | None = None,
+        user: str | None = None,
+        password: str | None = None,
+        database: str | None = None,
+        schema: str | None = None,
+        connection_params: dict[str, Any] | None = None,
+        fallback_stores: list["MetadataStore"] | None = None,
+        enable_pgcrypto: bool = True,
```

Comment:
Default should be False: we should not attempt to silently manage infrastructure 

We might want to have the same flag with DuckDB by the way, but maybe there it isn't that critical

Reply:
done; for duckdb - I think we do not necessarily need that flag as there it is easier to install the addons/more common. However, in the case of an airgapped installation things would indeed get tricky as there is not yet any (at least not to me known) artifactory for proxying the duckdb addon ecosystem